### PR TITLE
Read MC event info in Irene

### PR DESCRIPTION
**Pull request by jmbenlloch**
Irene now reads event number & timestamp from file both in MC and data.

There is also one detail, in MCRD and Diomira we have:
```
/Run                     Group
/Run/events              Dataset {10/Inf}
/Run/runInfo             Dataset {1}
```

while in Irene we have:
```
/RunEvtInfo              Group
/RunEvtInfo/eventInfo    Dataset {10/Inf}
/RunEvtInfo/runInfo      Dataset {10/Inf}
```

At some point we should put the same names in all cities.

### CODE REVIEW

File: invisible_cities/cities/irene_test.py
Comment (jacg) [inline invisible_cities/cities/irene_test.py]: Could we get rid of these magic numbers? There are a couple of 3s here, and a couple of 0s a bit higher up.

**Comment by jjgomezcadenas** [comment]
Can you go ahead and fix Irene so that the table is the same than the table in DIMIRA? I believe that this PR is the right place to do it!

**Comment by jacg** [comment]
Whilst we're here, could we deal with the `0` magic numbers a few lines above the ones that you removed?

**Comment by jmbenlloch** [comment]
That is just to read first row in both tables and checking the have the same content. I have added a variable for that number.

**Comment by jjgomezcadenas** [comment]
@jmbenlloch I am happy to approve when @jacg is happy. This PR is important to have unified the tables of DIOMIRA and IRENE (and to get rid of magics). @jacg let @jmbenlloch know of any new suggestion, since he will be off tomorrow for a few days.

**Review by jacg — APPROVED** [review]
Needs squashing before merging.

**Comment by jacg** [comment]
Don't forget to talk about the table unification in the squshed commit message!

**Comment by jmbenlloch** [comment]
Done!

**Comment by jacg** [comment]
How does this change of table names affect us in terms of data files *that have already been written*?

**Comment by jmbenlloch** [comment]
I don't think it is a problem since we are regenerating all files now. There is a small diomira production (200k evts) going on right now, after that I will run Irene and if there is now more problems, we will run eventually the whole production (4M evts)

**Comment by jjgomezcadenas** [comment]
Not seriously. The point of pushing all this PR work was that we are going to reprocess all Monte Carlo data and we are getting new data from the detector... this afternoon! Old data can be ignored (or reprocessed if needed).

**Comment by jacg** [comment]
> Old data can be ignored (or reprocessed if needed).

I'm just trying to have some awareness of where we stand in terms of having to regenerate, reprocess, or throw away old data, and how much it's costing. And how this cost is likely to grow, in case we feel the need to do something like this again!

>  we will run eventually the whole production (4M evts)

What does this mean in terms of CPU hours?

**Comment by jjgomezcadenas** [comment]
>I'm just trying to have some awareness of where we stand in terms of having to regenerate, reprocess, or throw away old data, and how much it's costing. And how this cost is likely to grow, in case we feel the need to do something like this again!

Old data is not very useful, we had a lot of problems with it. The most likely scenario is that we ignore it and carry on with new data. Reprocessing MC is common practice and in this case was mandatory, since we had to add the new MC tables.

**Comment by jmbenlloch** [comment]
> What does this mean in terms of CPU hours?

Around 800 hours for diomira and a little bit less for Irene.

**Comment by jacg** [comment]
> Around 800 hours for diomira and a little bit less for Irene.

Has anybody ever profiled any of this?

**Comment by jmbenlloch** [comment]
Well, I can tell you the times from last time. First we have to copy the input file from the network filesystem to the computing node, then run diomira and in the end copy the output. This are the times in seconds for jobs with 10k evts each, 400 jobs in total:

```
copy input file
avg: 40.8040201005,	 std: 50.7643762238

diomira
avg: 11647.7110553,	 std: 5665.84610538

copy output file
avg: 73.675879397,	 std: 96.8217705237

total
avg: 11762.1909548,	 std: 5711.30528197
```

So in the end, is more than the 800 hours I told you before (that was a quick estimation I underestimated the time per event a little bit).  Not all machines in the CERN cluster have the same CPU, in some machines jobs take longer than in others. For more details I guess one should profile diomira.

**Comment by jjgomezcadenas** [comment]
Last time I tried, Diomira run at 1 second per event and Irene a 0.5 second per event.

**Comment by jacg** [comment]
Yes, yes, but I mean actually *profile*, you know, with a profiler.

If we are running stuff for the order of 1000 hours, then it might be worthwhile to optimize things a bit.

If it's not holding us back so far, then I wouldn't prioritize it, but we should aim to do it sometime in the not too distant future.

**Comment by jjgomezcadenas** [comment]
I did profile both DIOMIRA and IRENE. In the case of DIOMIRA all the time was consumed by Scipy functions, which are already highly optimised, so I am not sure we can optimise much more. In the case of Irene I found the bottlenecks and re-wrote the functions in cython. I think that the performance was quite optimised but I am happy to look at the thing together again! Of course it would be nice if we can scratch some extra time!

**Comment by jacg** [comment]
> all the time was consumed by Scipy functions

That's encouraging

> which are already highly optimised, so I am not sure we can optimise much more.

Well, if (for example) you are calling some scipy function N<sup>2</sup> times where N times would do then that would

+ be consistent with the profile output you describe, **and**
+ still leave room for improvement.

We should aim to look at it sometime when we're not too pressed by other things.

**Comment by jjgomezcadenas** [comment]
Agreed! Also I don't have much experience with profiling, thus looking forward to the peras!

**Pull request merged**
